### PR TITLE
deploy: update CSI sidecars to latest versions available

### LINKED ISSUE #4419: Update csi sidecars to the latest available versions
# Describe the bug #

Update csi sidecars to the latest available versions

### PR DESCRIPTION
**Pull request by iPraveenParihar**
# Describe what this PR does #

Below sidecars are updated with latest available versions

csi-node-driver-registrar: v2.10.0
csi-resizer: v1.10.0
csi-provisioner: v4.0.0
csi-attacher: v4.5.0
csi-snapshotter: v7.0.0

## Related issues ##

Fixes: #4419 

**Checklist:**

* [x] **Commit Message Formatting**: Commit titles and messages follow guidelines in the [developer guide](https://github.com/ceph/ceph-csi/blob/devel/docs/development-guide.md#commit-messages).
* [x] Reviewed the developer guide on [Submitting a Pull Request](https://github.com/ceph/ceph-csi/blob/devel/docs/development-guide.md#development-workflow)
* [x] [Pending release notes](https://github.com/ceph/ceph-csi/blob/devel/PendingReleaseNotes.md) updated with breaking and/or notable changes for the next major release.
* [x] Documentation has been updated, if necessary.
* [x] Unit tests have been added, if necessary.
* [x] Integration tests have been added, if necessary.

---

<details>
<summary>Show available bot commands</summary>

These commands are normally not required, but in case of issues, leave any of
the following bot commands in an otherwise empty comment in this PR:

* `/retest ci/centos/<job-name>`: retest the `<job-name>` after unrelated
  failure (please report the failure too!)

</details>

### CODE REVIEW

File: build.env [hunk 1]
```diff
@@ -54,11 +54,11 @@ ROOK_VERSION=v1.12.1
...
 # CSI sidecar version
-CSI_ATTACHER_VERSION=v4.4.3
-CSI_SNAPSHOTTER_VERSION=v6.3.3
-CSI_RESIZER_VERSION=v1.9.3
-CSI_PROVISIONER_VERSION=v3.6.3
-CSI_NODE_DRIVER_REGISTRAR_VERSION=v2.9.3
+CSI_ATTACHER_VERSION=v4.5.0
+CSI_SNAPSHOTTER_VERSION=v7.0.0
+CSI_RESIZER_VERSION=v1.10.0
+CSI_PROVISIONER_VERSION=v4.0.0
```
Comment (Madhu-1) [inline build.env:60]: can you please check if any new RBAC changes are required for any new features as some of them are major releases?

Comment (iPraveenParihar) [reply build.env:60]: Sure

Comment (iPraveenParihar) [reply build.env:60]: > can you please check if any new RBAC changes are required for any new features as some of them are major releases?

Checked, there are no RBAC changes required.

**Comment by iPraveenParihar** [comment]
@Madhu-1, 
Now with latest provisioner we have `prevent-volume-mode-conversion` feature flag enabled by default.
we have it enabled explicitly, can we remove it?

https://github.com/ceph/ceph-csi/blob/141928d1532f289e76ac48947017d0162ef5f6ca/deploy/rbd/kubernetes/csi-rbdplugin-provisioner.yaml#L58-L60

**Comment by nixpanic** [comment]
> @Madhu-1, Now with latest provisioner we have `prevent-volume-mode-conversion` feature flag enabled by default. we have it enabled explicitly, can we remove it?
> 
> https://github.com/ceph/ceph-csi/blob/141928d1532f289e76ac48947017d0162ef5f6ca/deploy/rbd/kubernetes/csi-rbdplugin-provisioner.yaml#L58-L60

I think it makes sense to keep it as it is for now. In the future we may want to enable it for backup applications that use the [workflow described in the last section of this **Motivation**](https://github.com/kubernetes/enhancements/blob/master/keps/sig-storage/3141-prevent-volume-mode-conversion/README.md#motivation).

**Review by Madhu-1 — APPROVED** [review]

**Review by nixpanic — APPROVED** [review]

**Comment by nixpanic** [comment]
This can be queued for merging once #4422 is finished with CI.

**Comment by nixpanic** [comment]
@mergifyio rebase

**Comment by nixpanic** [comment]
@mergifyio queue

**Pull request merged**